hr_public_holidays/models/hr_public_holidays.py

```diff
@@ -0,0 +1,115 @@
+# -*- coding:utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2011,2013 Michael Telahun Makonnen <mmakonnen@gmail.com>.
+#    Copyright (C) 2014 initOS GmbH & Co. KG (<http://www.initos.com>).
+#    Author Nikolina Todorova <nikolina.todorova@initos.com>
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import models, fields, api, _
+from datetime import date, datetime
+from openerp.exceptions import ValidationError
+
+
+class HrPublicHolidays(models.Model):
+    _name = 'hr.holidays.public'
+    _description = 'Public Holidays'
+    _rec_name = 'year'
+    _order = 'year'
+
+    year = fields.Char("Calendar Year", required=True,
+                       help="Enter the year with a numeric value "
+                       "e.g. '2015' or '2016'.")
+    line_ids = fields.One2many('hr.holidays.public.line', 'holidays_id',
+                               'Holiday Dates')
+    country_id = fields.Many2one('res.country', 'Country')
+
+    @api.one
+    @api.constrains('year', 'country_id')
+    def _check_year(self):
+        try:
+            datetime.strptime(self.year, '%Y')
+        except:
+            raise ValidationError(
+                _("The year %s must be written with a numeric value "
+                  "e.g. '2015' or '2016'.") % self.year)
+        if not self.country_id:
+            pholidays = self.search([
+                ('year', '=', self.year),
+                ('country_id', '=', False),
+                ('id', '!=', self.id)])
+            if pholidays:
+                raise ValidationError(
+                    _('Error: Duplicate year %s') % self.year)
```

Comment:
This validation is not required (from line 50 to 57). It is checked by the unique constraint below.